controllers/etcd_controller_test.go

```diff
@@ -271,6 +340,367 @@ func podDeleted(c client.Client, etcd *druidv1alpha1.Etcd) error {
 
 }
 
+func validateEtcdConfigWithDefaults(s *appsv1.StatefulSet, cm *corev1.ConfigMap, svc *corev1.Service, instance *druidv1alpha1.Etcd) {
+	// Validate Quota
+	configYML := cm.Data[etcdConfig]
+	config := map[string]string{}
+	err := yaml.Unmarshal([]byte(configYML), &config)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(instance.Spec.Etcd.Quota).To(BeNil())
+	val, ok := config[quotaKey]
+	Expect(ok).To(BeTrue())
+	Expect(val).Should(Equal(fmt.Sprintf("%d", quota.Value())))
+
+	// Validate DefragmentationSchedule *string
+	Expect(instance.Spec.Etcd.DefragmentationSchedule).To(BeNil())
+	containers := map[string]corev1.Container{}
+	for _, container := range s.Spec.Template.Spec.Containers {
+		if container.Name == backupRestore {
+			containers[backupRestore] = container
+		} else if container.Name == common.Etcd {
+			containers[common.Etcd] = container
+		}
+	}
+
+	for _, commandParameter := range containers[backupRestore].Command {
+		if strings.HasPrefix(commandParameter, "--defragmentation-schedule") {
+			Fail("--defragmentation-schedule should not be present if etcd resource does not have DefragmentationSchedule")
+		}
+	}
+
+	// Validate ServerPort and ClientPort
+	for _, port := range containers[common.Etcd].Ports {
+		if port.Name == "server" {
+			Expect(instance.Spec.Etcd.ServerPort).To(BeNil())
+			Expect(port.ContainerPort).To(Equal(serverPort))
+		} else if port.Name == "client" {
+			Expect(instance.Spec.Etcd.ClientPort).To(BeNil())
+			Expect(port.ContainerPort).To(Equal(clientPort))
+		}
+	}
+	Expect(instance.Spec.Etcd.Image).To(BeNil())
+	imageVector, err := imagevector.ReadGlobalImageVectorWithEnvOverride(getImageYAMLPath())
+	Expect(err).NotTo(HaveOccurred())
+	images, err := imagevector.FindImages(imageVector, imageNames)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(containers[common.Etcd].Image).To(Equal(fmt.Sprintf("%s:%s", images[common.Etcd].Repository, *images[common.Etcd].Tag)))
```

Comment:
https://onsi.github.io/gomega/#consistofelement-interface?
https://onsi.github.io/gomega/#containelementelement-interface?

Reply:
It is just a string comparison here. Equal should suffice, ryt?